# Decomposing samples
MutaGene allows you to identify mutational signatures in samples. The following steps walk you through the collection of samples, reference genomes and decomposition of the samples using 

## 1. Collect the necessary files
[If you do not have sample or reference files click here](fetch.ipynb)

## 2. Decompose the samples
Now, we can use the `mutagene signature` command to process the MAF file and see the results.

- The option `-i` specifies the input file
- `-g` specifies the reference genome
- `-s` specifies the signature set
    - `5`: MutaGene 5
    - `10`: MutaGene 10
    - `30`: COSMIC v2
    - `49`: COSMIC v3
- `-f` specifies the input file format

In [23]:
!mutagene signature -i ./skcm_yale/data_mutations_mskcc.txt -g hg19.2bit -s 10 -f MAF

WARNING REF allele does not match the genomic sequence in chr6:24596477 T!=G. Multiple errors could mean wrong genome assembly choice
WARNING REF allele does not match the genomic sequence in chr20:60485626 C!=A. Multiple errors could mean wrong genome assembly choice
WARNING REF allele does not match the genomic sequence in chr3:3076442 A!=C. Multiple errors could mean wrong genome assembly choice
sample	signature	exposure	mutations
YURUS	1	0.0061	5
YURUS	5	0.0069	6
YURUS	6	0.0253	20
YURUS	7	0.0367	29
YURUS	9	0.8365	667
YURUS	10	0.0106	8
YUPAT	1	0.0408	40
YUPAT	2	0.0101	10
YUPAT	5	0.0090	9
YUPAT	6	0.0768	76
YUPAT	7	0.0368	36
YUPAT	8	0.0304	30
YUPAT	9	0.7314	719
YUPAT	10	0.0107	10
YUNACK	1	0.0462	5
YUNACK	2	0.0151	1
YUNACK	4	0.0605	6
YUNACK	5	0.0534	5
YUNACK	7	0.0301	3
YUNACK	8	0.0329	3
YUNACK	9	0.7091	70
YUWAND	1	0.0049	8
YUWAND	2	0.0020	3
YUWAND	4	0.0089	14
YUWAND	5	0.0084	13
YUWAND	6	0.0063	10
YUWAND	7	0.0035	5
YUWAND	8	0.0314	48
YUWAND	9	0.8085	1241
YUWAND	10	0.0092	14
YUGAFFE	1	0.

The output displays all the samples and their respective decompositions. For example:
```
sample	signature	exposure	mutations
YUWAND	1	0.0049	8
YUWAND	2	0.0020	3
YUWAND	4	0.0089	14
YUWAND	5	0.0084	13
YUWAND	6	0.0063	10
YUWAND	7	0.0035	5
YUWAND	8	0.0314	48
YUWAND	9	0.8085	1241
YUWAND	10	0.0092	14
```
In the case of sample YUWAND, for example, we can observe that 80.85% of mutations (corresponding to 1241 mutations) seem to be contributed by [signature 9](https://www.ncbi.nlm.nih.gov/research/mutagene/signatures/34268) of the [MutaGene 10 signature set](https://www.ncbi.nlm.nih.gov/research/mutagene/signatures#mutagene-10). This signature is associated with exposure to UV light.